tempto-kafka/src/main/java/io/prestodb/tempto/fulfillment/table/kafka/KafkaTableManager.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestodb.tempto.fulfillment.table.kafka;
+
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.name.Names;
+import io.prestodb.tempto.fulfillment.table.MutableTableRequirement;
+import io.prestodb.tempto.fulfillment.table.TableDefinition;
+import io.prestodb.tempto.fulfillment.table.TableHandle;
+import io.prestodb.tempto.fulfillment.table.TableInstance;
+import io.prestodb.tempto.fulfillment.table.TableManager;
+import io.prestodb.tempto.internal.fulfillment.table.TableName;
+import io.prestodb.tempto.query.QueryExecutor;
+import io.prestodb.tempto.query.QueryResult;
+import kafka.admin.AdminUtils;
+import kafka.admin.RackAwareMode;
+import kafka.utils.ZKStringSerializer$;
+import kafka.utils.ZkUtils;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.ZkConnection;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.slf4j.Logger;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.slf4j.LoggerFactory.getLogger;
+
+@TableManager.Descriptor(tableDefinitionClass = KafkaTableDefinition.class, type = "KAFKA")
+@Singleton
+public class KafkaTableManager
+        implements TableManager<KafkaTableDefinition>
+{
+    private static final Logger LOGGER = getLogger(KafkaTableManager.class);
+
+    private final String databaseName;
+    private final QueryExecutor prestoQueryExecutor;
+    private final String brokerHost;
+    private final Integer brokerPort;
+    private final String prestoKafkaCatalog;
+    private final String zookeeperHost;
+    private final Integer zookeeperPort;
+
+    @Inject
+    public KafkaTableManager(
+            @Named("databaseName") String databaseName,
+            @Named("brokerHost") String brokerHost,
+            @Named("brokerPort") int brokerPort,
+            @Named("zookeeperHost") String zookeeperHost,
+            @Named("zookeeperPort") int zookeeperPort,
+            @Named("prestoDatabaseName") String prestoDatabaseName,
+            @Named("prestoKafkaCatalog") String prestoKafkaCatalog,
```

Comment:
It seems to be first time that we are using camel case. Why not things like `zookeeper.host`?

Reply:
`zookeeper.host` would imply deeper nesting in yaml. We do not do that for other props. I will make it `zookeeper_host` for consistency with other params though.